### ISSUE TITLE
HBLANK is good?

### ISSUE THREAD

**Issue by gyurco** [ika-musume/ikacore_SuprLoco#1]
There are scrolling artifacts at the left and right side of the screen with the hblank in the core. According to MAME, it's a 240x224 game, and the core has a horizontal resoultion of 256 pixels. Seems the blank output from the PAL is only consist of the VBLANK, and HBLANK is generated elsewhere, so I changed it to:
```
    if (hcntr == 8'hf9) o_HBLANK_n <= 0; else if (hcntr == 8'h09) o_HBLANK_n <= 1;
```
Looks better for me (from hFA to h0A = 240 pixels). I don't know how it's done originally, the bottom PAL on the schematics in docs/ doesn't has either hcntr nor hblank as input, so nBLNK is really VBLANK only instead of composite blank? Or the PALs are not 100% reverse engineered?

**Comment by ika-musume** [comment]
Hi. 256px horizontal width is specification of the actual PCB. Here is  [a video](https://youtu.be/Bvmg1Ohk11I) I recorded on the board I have. You can modify the composite blank signal to mask the garbage graphics on either side of the screen. I didn't want to do that cuz that's the behavior of the real hardware. However, if someone send a pull request with mask feature to the [MiSTer-devel repo](https://github.com/MiSTer-devel/Arcade-SuprLoco_MiSTer), I'd be happy to accept it.

**Comment by gyurco** [comment]
I see. I think the garbage is worse with the core than on the video, so masking is more useful.
Unfortunately I ported it to other FPGA, I don't use MiSTer, so I cannot send a MiSTer specific PR.

**Comment by ika-musume** [comment]
It could also be because I started recording in the middle of the game. Compared to my PCB, the amount of garbage graphics didn't seem that high.

And when I compiled the Verilog file from the lower PAL into a jed file with ispLever and applied it to the actual PCB, the sprites got scrambled. The original PAL is DMPAL16R4 and I used ATF16V8. Sprites look fine on an FPGA anyway, so maybe it's simply a timing difference of modern, faster device.

The tilemap is 256px width and the sprite buffer is 240px width. The sprites actually have a buffer for 256 pixels, but for some reason it have to subtract 16 pixels for setup time(the buffer read counter can only display 240 pixels) They fixed this problem on Sega System 1. Sega seems to have programmed it thinking the game was going to have a horizontal resolution of 240px, so there shouldn't be any problem if you cover it up.
